### ISSUE TITLE
Bug: [prefer-optional-chain] Unsoundly fixes `a !== undefined && a.b !== null` to `a?.b !== null`

### ISSUE THREAD

**Issue by andersk** [typescript-eslint/typescript-eslint#10484]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.6.3&fileType=.tsx&code=GYVwdgxgLglg9mABMAFAQwPwC5EG9EBGOAzlAE4xgDmiAPomCADZOIC%2BAlDgG5wwAmeAFCJEMYInSIAhAF5ZicPwCmwSssEAyTYjQA6AjPkNmTDsNGiICYnCbK9TOFXQGOI9kLZCgA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Y6RAM0WloHsalfkwCG8WmQAWo5uii9o-aJHBgAviHVA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eFYDAruuGAL4g9A&tokens=false

### Repro Code

```TypeScript
function f(a?: { b: string | null }): void {
  if (a !== undefined && a.b !== null) {
    console.log(a.b)
  }
}
```

### ESLint Config

```javascript
module.exports = {
  "rules": {
    "@typescript-eslint/prefer-optional-chain": "error"
  }
}
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "strict": true
  }
}
```

### Expected Result

No suggestion.

### Actual Result

prefer-optional-chain suggests this fix:

```diff
 function f(a?: { b: string | null }): void {
-  if (a !== undefined && a.b !== null) {
+  if (a?.b !== null) {
     console.log(a.b)
   }
 }
```

This is unsound because if `a` is `undefined`, then `a?.b` is `undefined`, not `null`.

**Comment by bradzacher** [comment]
In this instance the rule has a _suggestion_ fixer -- a fixer that the user must manually apply  -- and not an _auto_fixer -- a fixer that is applied automatically via `--fix`).

Suggestion fixers are not required to be sound, fully correct, or even fully complete and instead are a tool to provide starting point for the user to fix the report. When manually applying a suggestion fixer -- it is up to the user to review the code and correct it if needs be.

That being said -- it might be a weird edge case where the fixer should behave differently?
I dunno -- the code is extremely complicated and cursed cos even suggesting fixers for this stuff is disgusting 🤮.

**Comment by kirkwaiblinger** [comment]
I personally wouldn't hesitate to write it like
```diff
function f(a?: { b: string | null }): void {
-  if (a !== undefined && a.b !== null) {
+  if (a?.b != null) {
    console.log(a.b)
  }
}
```
but a) communicating out the benefits of loose nullish equality is tricky and controversial, and b) that strategy isn't sound for the following, where I _really_ don't think optional chaining is possible to use (since `a === undefined` and `a.b === undefined` need to have different behavior
```ts
function f(a?: { b: string | null | undefined }): void {
  if (a !== undefined && a.b !== null) {
    console.log(a.b)
  }
}
```

**Comment by kirkwaiblinger** [comment]
> b) that strategy isn't sound for the following, where I _really_ don't think optional chaining is possible to use (since `a === undefined` and `a.b === undefined` need to have different behavior
> 
> ```ts
> function f(a?: { b: string | null | undefined }): void {
>   if (a !== undefined && a.b !== null) {
>     console.log(a.b)
>   }
> }
> ```

Oh - but the rule detects this and doesn't report in that case. So I'm leaning towards thinking we should try to handle this with (loose) `!= null` in the suggestion fixer, or not report at all if we don't want to engage in the `!= null` vs `!== null` conversation.
